rust/src/connection.rs

```diff
@@ -0,0 +1,56 @@
+use crate::config;
+use crate::wallet;
+use anyhow::Result;
+
+/// blocks the current thread, if you want this to be run asynchronously you need to spawn a thread
+/// from the outside.
+pub async fn keep_alive() -> Result<()> {
+    let mut connected = false;
+    loop {
+        let alive = check_alive().await?;
+
+        if !connected || !alive {
```

Comment:
I'm not sure this makes 100% sense, help me out please :) 

- if `alive=false` the maker's http is not reachable but we try to connect via LDK 
- if `alive=true` the maker's http is reachable (which should always be the case), but we could have lost the connection the his LDK and will never reconnect

Reply:
It's the same logic as before this change though. 

I agree that we should find a way to notice that the LDK connection is dead, but perhaps in a separate PR?

Reply:
@bonomat you are completely right. like mentioned in the last PR, it would be much nicer if we handle the actual event instead of checking if the maker is alive. This was a workaround - this PR is only a small clean up to not block another thread. I am creating a ticket for the handling the actual event - https://github.com/itchysats/10101/issues/433